trilinos@develop: kokkos_tpls.cmake: Enabled TPL HWLOC inside TriBITS build, but this can only be enabled in a standalone build

### Steps to reproduce the issue

The following spec fails to build:
```
trilinos@develop%gcc@11.2.0~adelus~adios2+amesos+amesos2+anasazi+aztec~basker+belos+boost~chaco~complex~cuda~cuda_rdc~debug~dtk+epetra+epetraext~epetraextbtf~epetraextexperimental~epetraextgraphreorderings~exodus+explicit_template_instantiation~float+fortran~gtest~hdf5~hypre+ifpack+ifpack2+intrepid+intrepid2~ipo+isorropia+kokkos~mesquite+minitensor+ml+mpi+muelu~mumps+nox~openmp~panzer+phalanx+piro~python+rocm~rocm_rdc+rol+rythmos+sacado~scorec+shards+shared+shylu+stk+stokhos+stratimikos~strumpack~suite-sparse~superlu+superlu-dist+teko+tempus+thyra+tpetra+trilinoscouplings~wrapper~x11+zoltan+zoltan2 amdgpu_target=gfx90a build_system=cmake build_type=RelWithDebInfo cxxstd=17 generator=make gotype=long_long arch=linux-ubuntu20.04-x86_64
```

... using:
* `spack@develop` (3c3a4c75776ece43c95df46908dea026ac2a9276 from `Mon May 1 15:06:10 2023 +0200`)
* GCC 11.2.0 / Ubuntu 20.04
* ROCm 5.4.3

Concrete spec: [trilinos-develop.spec.json.txt](https://github.com/spack/spack/files/11453321/trilinos-develop.spec.json.txt)

<details><summary>Concretization</summary>
<pre>
 -   trilinos@develop%gcc@11.2.0~adelus~adios2+amesos+amesos2+anasazi+aztec~basker+belos+boost~chaco~complex~cuda~cuda_rdc~debug~dtk+epetra+epetraext~epetraextbtf~epetraextexperimental~epetraextgraphreorderings~exodus+explicit_template_instantiation~float+fortran~gtest~hdf5~hypre+ifpack+ifpack2+intrepid+intrepid2~ipo+isorropia+kokkos~mesquite+minitensor+ml+mpi+muelu~mumps+nox~openmp~panzer+phalanx+piro~python+rocm~rocm_rdc+rol+rythmos+sacado~scorec+shards+shared+shylu+stk+stokhos+stratimikos~strumpack~suite-sparse~superlu+superlu-dist+teko+tempus+thyra+tpetra+trilinoscouplings~wrapper~x11+zoltan+zoltan2 amdgpu_target=gfx90a build_system=cmake build_type=RelWithDebInfo cxxstd=17 generator=make gotype=long_long arch=linux-ubuntu20.04-x86_64
[+]      ^boost@1.82.0%gcc@11.2.0~atomic~chrono~clanglibcpp~container~context~contract~coroutine~date_time~debug+exception~fiber~filesystem+graph~graph_parallel~icu~iostreams~json~locale~log+math~mpi+multithreaded~nowide~numpy~pic~program_options~python~random~regex~serialization+shared~signals~singlethreaded+stacktrace~system~taggedlayout~test~thread~timer~type_erasure~versionedlayout~wave build_system=generic cxxstd=17 patches=a440f96,a7c807f visibility=hidden arch=linux-ubuntu20.04-x86_64
[+]      ^cmake@3.26.3%gcc@11.2.0~doc+ncurses+ownlibs~qt build_system=generic build_type=Release arch=linux-ubuntu20.04-x86_64
[+]          ^ncurses@6.4%gcc@11.2.0~symlinks+termlib abi=none build_system=autotools arch=linux-ubuntu20.04-x86_64
[+]          ^openssl@1.1.1t%gcc@11.2.0~docs~shared build_system=generic certs=mozilla arch=linux-ubuntu20.04-x86_64
[+]              ^ca-certificates-mozilla@2023-01-10%gcc@11.2.0 build_system=generic arch=linux-ubuntu20.04-x86_64
[+]      ^gmake@4.4.1%gcc@11.2.0~guile build_system=autotools arch=linux-ubuntu20.04-x86_64
[+]      ^hip@5.4.3%gcc@11.2.0~cuda~ipo+rocm build_system=cmake build_type=Release generator=make patches=ca523f1 arch=linux-ubuntu20.04-x86_64
[+]      ^hsa-rocr-dev@5.4.3%gcc@11.2.0+image~ipo+shared build_system=cmake build_type=Release generator=make patches=71e6851 arch=linux-ubuntu20.04-x86_64
[+]      ^hwloc@2.9.1%gcc@11.2.0~cairo~cuda~gl~libudev+libxml2~netloc~nvml~oneapi-level-zero~opencl+pci~rocm build_system=autotools libs=shared,static arch=linux-ubuntu20.04-x86_64
[+]          ^libpciaccess@0.17%gcc@11.2.0 build_system=autotools arch=linux-ubuntu20.04-x86_64
[+]              ^libtool@2.4.7%gcc@11.2.0 build_system=autotools arch=linux-ubuntu20.04-x86_64
[+]              ^util-macros@1.19.3%gcc@11.2.0 build_system=autotools arch=linux-ubuntu20.04-x86_64
[+]          ^libxml2@2.10.3%gcc@11.2.0~python build_system=autotools arch=linux-ubuntu20.04-x86_64
[+]              ^libiconv@1.17%gcc@11.2.0 build_system=autotools libs=shared,static arch=linux-ubuntu20.04-x86_64
[+]              ^xz@5.4.1%gcc@11.2.0~pic build_system=autotools libs=shared,static arch=linux-ubuntu20.04-x86_64
[+]          ^pkgconf@1.8.0%gcc@11.2.0 build_system=autotools arch=linux-ubuntu20.04-x86_64
[+]      ^llvm-amdgpu@5.4.3%gcc@11.2.0~ipo~link_llvm_dylib~llvm_dylib~openmp+rocm-device-libs build_system=cmake build_type=Release generator=ninja patches=a08bbe1 arch=linux-ubuntu20.04-x86_64
[+]      ^metis@5.1.0%gcc@11.2.0~gdb~int64~ipo~real64+shared build_system=cmake build_type=RelWithDebInfo generator=make patches=4991da9,93a7903,b1225da arch=linux-ubuntu20.04-x86_64
[+]      ^mpich@4.1.1%gcc@11.2.0~argobots~cuda+fortran+hwloc+hydra+libxml2+pci~rocm+romio~slurm~two_level_namespace~vci~verbs~wrapperrpath build_system=autotools datatype-engine=auto device=ch4 netmod=ofi pmi=pmi arch=linux-ubuntu20.04-x86_64
[+]          ^findutils@4.9.0%gcc@11.2.0 build_system=autotools patches=440b954 arch=linux-ubuntu20.04-x86_64
[+]          ^libfabric@1.18.0%gcc@11.2.0~debug~kdreg build_system=autotools fabrics=sockets,tcp,udp arch=linux-ubuntu20.04-x86_64
[+]          ^yaksa@0.2%gcc@11.2.0~cuda~rocm build_system=autotools arch=linux-ubuntu20.04-x86_64
[+]              ^autoconf@2.69%gcc@11.2.0 build_system=autotools patches=35c4492,7793209,a49dd5b arch=linux-ubuntu20.04-x86_64
[+]              ^automake@1.16.5%gcc@11.2.0 build_system=autotools arch=linux-ubuntu20.04-x86_64
[+]              ^m4@1.4.19%gcc@11.2.0+sigsegv build_system=autotools patches=9dc5fbd,bfdffa7 arch=linux-ubuntu20.04-x86_64
[+]                  ^diffutils@3.9%gcc@11.2.0 build_system=autotools arch=linux-ubuntu20.04-x86_64
[+]                  ^libsigsegv@2.14%gcc@11.2.0 build_system=autotools arch=linux-ubuntu20.04-x86_64
[+]              ^python@3.10.10%gcc@11.2.0+bz2+crypt+ctypes+dbm~debug+libxml2+lzma~nis~optimizations+pic+pyexpat+pythoncmd+readline+shared+sqlite3+ssl~tkinter+uuid+zlib build_system=generic patches=0d98e93,7d40923,f2fd060 arch=linux-ubuntu20.04-x86_64
[+]                  ^expat@2.5.0%gcc@11.2.0+libbsd build_system=autotools arch=linux-ubuntu20.04-x86_64
[+]                      ^libbsd@0.11.7%gcc@11.2.0 build_system=autotools arch=linux-ubuntu20.04-x86_64
[+]                          ^libmd@1.0.4%gcc@11.2.0 build_system=autotools arch=linux-ubuntu20.04-x86_64
[+]                  ^gettext@0.21.1%gcc@11.2.0+bzip2+curses+git~libunistring+libxml2+tar+xz build_system=autotools arch=linux-ubuntu20.04-x86_64
[+]                      ^tar@1.34%gcc@11.2.0 build_system=autotools zip=pigz arch=linux-ubuntu20.04-x86_64
[+]                          ^pigz@2.7%gcc@11.2.0 build_system=makefile arch=linux-ubuntu20.04-x86_64
[+]                          ^zstd@1.5.5%gcc@11.2.0+programs build_system=makefile compression=none libs=shared,static arch=linux-ubuntu20.04-x86_64
[+]                  ^libffi@3.4.4%gcc@11.2.0 build_system=autotools arch=linux-ubuntu20.04-x86_64
[+]                  ^libxcrypt@4.4.33%gcc@11.2.0~obsolete_api build_system=autotools arch=linux-ubuntu20.04-x86_64
[+]                  ^readline@8.2%gcc@11.2.0 build_system=autotools patches=bbf97f1 arch=linux-ubuntu20.04-x86_64
[+]                  ^sqlite@3.40.1%gcc@11.2.0+column_metadata+dynamic_extensions+fts~functions+rtree build_system=autotools arch=linux-ubuntu20.04-x86_64
[+]                  ^util-linux-uuid@2.38.1%gcc@11.2.0 build_system=autotools arch=linux-ubuntu20.04-x86_64
[+]      ^openblas@0.3.23%gcc@11.2.0~bignuma~consistent_fpcsr+fortran~ilp64+locking+pic+shared build_system=makefile symbol_suffix=none threads=none arch=linux-ubuntu20.04-x86_64
[+]          ^perl@5.36.0%gcc@11.2.0+cpanm+open+shared+threads build_system=generic arch=linux-ubuntu20.04-x86_64
[+]              ^berkeley-db@18.1.40%gcc@11.2.0+cxx~docs+stl build_system=autotools patches=26090f4,b231fcc arch=linux-ubuntu20.04-x86_64
[+]              ^bzip2@1.0.8%gcc@11.2.0~debug~pic+shared build_system=generic arch=linux-ubuntu20.04-x86_64
[+]              ^gdbm@1.23%gcc@11.2.0 build_system=autotools arch=linux-ubuntu20.04-x86_64
[+]      ^parmetis@4.0.3%gcc@11.2.0~gdb~int64~ipo+shared build_system=cmake build_type=RelWithDebInfo generator=make patches=4f89253,50ed208,704b84f arch=linux-ubuntu20.04-x86_64
[+]      ^superlu-dist@develop%gcc@11.2.0~cuda~int64~ipo~openmp~rocm+shared build_system=cmake build_type=RelWithDebInfo generator=make arch=linux-ubuntu20.04-x86_64
[+]      ^zlib@1.2.13%gcc@11.2.0+optimize+pic+shared build_system=makefile arch=linux-ubuntu20.04-x86_64
</pre></details>

Error:
```
$> spack install -f ./trilinos-develop.spec.json
...
==> Installing trilinos-develop-6pqbqnx4d3tq3gjfqvjri5hn7ubivtdw
==> No binary for trilinos-develop-6pqbqnx4d3tq3gjfqvjri5hn7ubivtdw found: installing from source
==> No patches needed for trilinos
==> trilinos: Executing phase: 'cmake'
==> Error: ProcessError: Command exited with status 1:
    '/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.2.0/cmake-3.26.3-i2vcdcd5s6tu5bvwvbkx57iuyr6sip63/bin/cmake' '-G' 'Unix Makefiles' '-DCMAKE_INSTALL_PREFIX:STRING=/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.2.0/trilinos-develop-6pqbqnx4d3tq3gjfqvjri5hn7ubivtdw' '-DCMAKE_BUILD_TYPE:STRING=RelWithDebInfo' '-DBUILD_TESTING:BOOL=OFF' '-DCMAKE_INTERPROCEDURAL_OPTIMIZATION:BOOL=OFF' '-DCMAKE_VERBOSE_MAKEFILE:BOOL=ON' '-DCMAKE_INSTALL_RPATH_USE_LINK_PATH:BOOL=ON' '-DCMAKE_INSTALL_RPATH:STRING=/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.2.0/trilinos-develop-6pqbqnx4d3tq3gjfqvjri5hn7ubivtdw/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.2.0/trilinos-develop-6pqbqnx4d3tq3gjfqvjri5hn7ubivtdw/lib64;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.2.0/boost-1.82.0-tdw2rfmq2qs3i7yvzf6knnwuecblozip/lib;/opt/rocm-5.4.3/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.2.0/hwloc-2.9.1-qf3i4i5rbanuj7sw6trgnw3h575f35gu/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.2.0/libpciaccess-0.17-t3s3waig7ovxllcidsntd4uozl5dj23m/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.2.0/libxml2-2.10.3-5dkkc5mmka4i56bztbbslmn3pxxdlddr/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.2.0/libiconv-1.17-m3pvej7psgqle3np2lvba64uvoea2csg/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.2.0/xz-5.4.1-ca7k7h4uyhw55pacq5kfrbrq7e4cero3/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.2.0/zlib-1.2.13-4a4hcyyb5o4jztxdzjxzu2sri67xebjf/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.2.0/ncurses-6.4-w5d2qa23mw4wccvrrzs2ggwvn7fnuha7/lib;/opt/rocm-5.4.3/llvm/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.2.0/metis-5.1.0-cfyxpccu64h64r2nglgvxofktyitdspg/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.2.0/mpich-4.1.1-una3gsqpqddps5po56yrpbgyncipegdx/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.2.0/libfabric-1.18.0-po72vtkddvx6s2knrmnzbphibltrlq6q/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.2.0/yaksa-0.2-gd6xmctsszzsawps5gisuevvpwvisd3z/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.2.0/openblas-0.3.23-rfka6emvjf67auwa2dwzh4mx757ewmcp/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.2.0/parmetis-4.0.3-6kzi6romat5pn7jpmqz2v3v4irkr75s4/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.2.0/superlu-dist-develop-7y5yg4m7usuamo6dmgxy547doknrs5rf/lib' '-DCMAKE_PREFIX_PATH:STRING=/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.2.0/superlu-dist-develop-7y5yg4m7usuamo6dmgxy547doknrs5rf;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.2.0/parmetis-4.0.3-6kzi6romat5pn7jpmqz2v3v4irkr75s4;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.2.0/openblas-0.3.23-rfka6emvjf67auwa2dwzh4mx757ewmcp;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.2.0/mpich-4.1.1-una3gsqpqddps5po56yrpbgyncipegdx;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.2.0/yaksa-0.2-gd6xmctsszzsawps5gisuevvpwvisd3z;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.2.0/libfabric-1.18.0-po72vtkddvx6s2knrmnzbphibltrlq6q;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.2.0/metis-5.1.0-cfyxpccu64h64r2nglgvxofktyitdspg;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.2.0/hwloc-2.9.1-qf3i4i5rbanuj7sw6trgnw3h575f35gu;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.2.0/libxml2-2.10.3-5dkkc5mmka4i56bztbbslmn3pxxdlddr;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.2.0/xz-5.4.1-ca7k7h4uyhw55pacq5kfrbrq7e4cero3;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.2.0/libpciaccess-0.17-t3s3waig7ovxllcidsntd4uozl5dj23m;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.2.0/gmake-4.4.1-bdi4lzxyoq3s53nmmnnmd6xkedbl5mt7;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.2.0/cmake-3.26.3-i2vcdcd5s6tu5bvwvbkx57iuyr6sip63;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.2.0/zlib-1.2.13-4a4hcyyb5o4jztxdzjxzu2sri67xebjf;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.2.0/libiconv-1.17-m3pvej7psgqle3np2lvba64uvoea2csg;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.2.0/ncurses-6.4-w5d2qa23mw4wccvrrzs2ggwvn7fnuha7;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.2.0/boost-1.82.0-tdw2rfmq2qs3i7yvzf6knnwuecblozip;/opt/rocm-5.4.3/llvm;/opt/rocm-5.4.3/;/opt/rocm-5.4.3' '-DTrilinos_VERBOSE_CONFIGURE:BOOL=OFF' '-DBUILD_SHARED_LIBS:BOOL=ON' '-DTrilinos_ENABLE_ALL_OPTIONAL_PACKAGES:BOOL=OFF' '-DTrilinos_ENABLE_ALL_PACKAGES:BOOL=OFF' '-DTrilinos_ENABLE_CXX11:BOOL=ON' '-DTrilinos_ENABLE_DEBUG:BOOL=OFF' '-DTrilinos_ENABLE_EXAMPLES:BOOL=OFF' '-DTrilinos_ENABLE_SECONDARY_TESTED_CODE:BOOL=ON' '-DTrilinos_ENABLE_TESTS:BOOL=OFF' '-DTrilinos_ENABLE_Fortran:BOOL=ON' '-DTrilinos_ENABLE_OpenMP:BOOL=OFF' '-DTrilinos_ENABLE_EXPLICIT_INSTANTIATION:BOOL=ON' '-DCMAKE_CXX_STANDARD:STRING=17' '-DTrilinos_ENABLE_Adelus:BOOL=OFF' '-DTrilinos_ENABLE_Amesos:BOOL=ON' '-DTrilinos_ENABLE_Amesos2:BOOL=ON' '-DTrilinos_ENABLE_Anasazi:BOOL=ON' '-DTrilinos_ENABLE_AztecOO:BOOL=ON' '-DTrilinos_ENABLE_Belos:BOOL=ON' '-DTrilinos_ENABLE_Epetra:BOOL=ON' '-DTrilinos_ENABLE_EpetraExt:BOOL=ON' '-DTrilinos_ENABLE_FEI:BOOL=OFF' '-DTrilinos_ENABLE_Gtest:BOOL=OFF' '-DTrilinos_ENABLE_Ifpack:BOOL=ON' '-DTrilinos_ENABLE_Ifpack2:BOOL=ON' '-DTrilinos_ENABLE_Intrepid:BOOL=ON' '-DTrilinos_ENABLE_Intrepid2:BOOL=ON' '-DTrilinos_ENABLE_Isorropia:BOOL=ON' '-DTrilinos_ENABLE_Kokkos:BOOL=ON' '-DTrilinos_ENABLE_MiniTensor:BOOL=ON' '-DTrilinos_ENABLE_Mesquite:BOOL=OFF' '-DTrilinos_ENABLE_ML:BOOL=ON' '-DTrilinos_ENABLE_MueLu:BOOL=ON' '-DTrilinos_ENABLE_NOX:BOOL=ON' '-DTrilinos_ENABLE_Pamgen:BOOL=OFF' '-DTrilinos_ENABLE_Panzer:BOOL=OFF' '-DTrilinos_ENABLE_Pike:BOOL=OFF' '-DTrilinos_ENABLE_Piro:BOOL=ON' '-DTrilinos_ENABLE_Phalanx:BOOL=ON' '-DTrilinos_ENABLE_PyTrilinos:BOOL=OFF' '-DTrilinos_ENABLE_ROL:BOOL=ON' '-DTrilinos_ENABLE_Rythmos:BOOL=ON' '-DTrilinos_ENABLE_Sacado:BOOL=ON' '-DTrilinos_ENABLE_SCOREC:BOOL=OFF' '-DTrilinos_ENABLE_Shards:BOOL=ON' '-DTrilinos_ENABLE_ShyLU:BOOL=ON' '-DTrilinos_ENABLE_STK:BOOL=ON' '-DTrilinos_ENABLE_Stokhos:BOOL=ON' '-DTrilinos_ENABLE_Stratimikos:BOOL=ON' '-DTrilinos_ENABLE_Teko:BOOL=ON' '-DTrilinos_ENABLE_Tempus:BOOL=ON' '-DTrilinos_ENABLE_Thyra:BOOL=ON' '-DTrilinos_ENABLE_Tpetra:BOOL=ON' '-DTrilinos_ENABLE_TrilinosCouplings:BOOL=ON' '-DTrilinos_ENABLE_Triutils:BOOL=ON' '-DTrilinos_ENABLE_Zoltan:BOOL=ON' '-DTrilinos_ENABLE_Zoltan2:BOOL=ON' '-DEpetraExt_BUILD_BTF:BOOL=OFF' '-DEpetraExt_BUILD_EXPERIMENTAL:BOOL=OFF' '-DEpetraExt_BUILD_GRAPH_REORDERINGS:BOOL=OFF' '-DAmesos2_ENABLE_Basker:BOOL=OFF' '-DAmesos2_ENABLE_LAPACK:BOOL=ON' '-DAmesos2_ENABLE_MUMPS:BOOL=OFF' '-DTrilinos_ENABLE_SEACASExodus:BOOL=OFF' '-DTrilinos_ENABLE_SEACASIoss:BOOL=OFF' '-DTrilinos_ENABLE_SEACASChaco:BOOL=OFF' '-DTrilinos_ENABLE_SEACASNemslice:BOOL=OFF' '-DTrilinos_ENABLE_ThyraEpetraAdapters:BOOL=ON' '-DTrilinos_ENABLE_ThyraEpetraExtAdapters:BOOL=ON' '-DTrilinos_ENABLE_ThyraTpetraAdapters:BOOL=ON' '-DTPL_ENABLE_ADIOS2:BOOL=OFF' '-DTPL_ENABLE_Boost:BOOL=ON' '-DBoost_INCLUDE_DIRS:STRING=/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.2.0/boost-1.82.0-tdw2rfmq2qs3i7yvzf6knnwuecblozip/include' '-DBoost_ROOT:STRING=/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.2.0/boost-1.82.0-tdw2rfmq2qs3i7yvzf6knnwuecblozip' '-DBoost_LIBRARY_NAMES:STRING=boost_graph;boost_graph-mt;boost_math_c99;boost_math_tr1f-mt;boost_math_tr1l-mt;boost_math_c99f-mt;boost_math_tr1-mt;boost_math_c99l-mt;boost_math_c99-mt;boost_math_c99f;boost_math_tr1;boost_math_c99l;boost_math_tr1f;boost_math_tr1l;boost_stacktrace_noop-mt;boost_stacktrace_basic;boost_stacktrace_noop;boost_stacktrace_addr2line;boost_stacktrace_addr2line-mt;boost_stacktrace_basic-mt' '-DBoost_LIBRARY_DIRS:STRING=/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.2.0/boost-1.82.0-tdw2rfmq2qs3i7yvzf6knnwuecblozip/lib' '-DTPL_ENABLE_CUDA:BOOL=OFF' '-DTPL_ENABLE_HDF5:BOOL=OFF' '-DTPL_ENABLE_HYPRE:BOOL=OFF' '-DTPL_ENABLE_MUMPS:BOOL=OFF' '-DTPL_ENABLE_UMFPACK:BOOL=OFF' '-DTPL_ENABLE_SuperLU:BOOL=OFF' '-DTPL_ENABLE_SuperLUDist:BOOL=ON' '-DSuperLUDist_INCLUDE_DIRS:STRING=/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.2.0/superlu-dist-develop-7y5yg4m7usuamo6dmgxy547doknrs5rf/include' '-DSuperLUDist_ROOT:STRING=/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.2.0/superlu-dist-develop-7y5yg4m7usuamo6dmgxy547doknrs5rf' '-DSuperLUDist_LIBRARY_NAMES:STRING=superlu_dist' '-DSuperLUDist_LIBRARY_DIRS:STRING=/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.2.0/superlu-dist-develop-7y5yg4m7usuamo6dmgxy547doknrs5rf/lib' '-DTPL_ENABLE_X11:BOOL=OFF' '-DTPL_ENABLE_STRUMPACK:BOOL=OFF' '-DTPL_ENABLE_BLAS:BOOL=ON' '-DBLAS_INCLUDE_DIRS:STRING=/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.2.0/openblas-0.3.23-rfka6emvjf67auwa2dwzh4mx757ewmcp/include' '-DBLAS_ROOT:STRING=/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.2.0/openblas-0.3.23-rfka6emvjf67auwa2dwzh4mx757ewmcp' '-DBLAS_LIBRARY_NAMES:STRING=openblas' '-DBLAS_LIBRARY_DIRS:STRING=/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.2.0/openblas-0.3.23-rfka6emvjf67auwa2dwzh4mx757ewmcp/lib' '-DTPL_ENABLE_CGNS:BOOL=OFF' '-DTPL_ENABLE_LAPACK:BOOL=ON' '-DLAPACK_INCLUDE_DIRS:STRING=/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.2.0/openblas-0.3.23-rfka6emvjf67auwa2dwzh4mx757ewmcp/include' '-DLAPACK_ROOT:STRING=/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.2.0/openblas-0.3.23-rfka6emvjf67auwa2dwzh4mx757ewmcp' '-DLAPACK_LIBRARY_NAMES:STRING=openblas' '-DLAPACK_LIBRARY_DIRS:STRING=/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.2.0/openblas-0.3.23-rfka6emvjf67auwa2dwzh4mx757ewmcp/lib' '-DTPL_ENABLE_Matio:BOOL=OFF' '-DTPL_ENABLE_METIS:BOOL=ON' '-DMETIS_INCLUDE_DIRS:STRING=/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.2.0/metis-5.1.0-cfyxpccu64h64r2nglgvxofktyitdspg/include' '-DMETIS_ROOT:STRING=/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.2.0/metis-5.1.0-cfyxpccu64h64r2nglgvxofktyitdspg' '-DMETIS_LIBRARY_NAMES:STRING=metis' '-DMETIS_LIBRARY_DIRS:STRING=/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.2.0/metis-5.1.0-cfyxpccu64h64r2nglgvxofktyitdspg/lib' '-DTPL_ENABLE_Netcdf:BOOL=OFF' '-DTPL_ENABLE_SCALAPACK:BOOL=OFF' '-DTPL_ENABLE_Zlib:BOOL=ON' '-DZlib_INCLUDE_DIRS:STRING=/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.2.0/zlib-1.2.13-4a4hcyyb5o4jztxdzjxzu2sri67xebjf/include' '-DZlib_ROOT:STRING=/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.2.0/zlib-1.2.13-4a4hcyyb5o4jztxdzjxzu2sri67xebjf' '-DZlib_LIBRARY_NAMES:STRING=z' '-DZlib_LIBRARY_DIRS:STRING=/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.2.0/zlib-1.2.13-4a4hcyyb5o4jztxdzjxzu2sri67xebjf/lib' '-DTPL_ENABLE_Pnetcdf:BOOL=OFF' '-DTPL_ENABLE_HWLOC:BOOL=ON' '-DHWLOC_INCLUDE_DIRS:STRING=/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.2.0/hwloc-2.9.1-qf3i4i5rbanuj7sw6trgnw3h575f35gu/include' '-DHWLOC_ROOT:STRING=/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.2.0/hwloc-2.9.1-qf3i4i5rbanuj7sw6trgnw3h575f35gu' '-DHWLOC_LIBRARY_NAMES:STRING=hwloc' '-DHWLOC_LIBRARY_DIRS:STRING=/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.2.0/hwloc-2.9.1-qf3i4i5rbanuj7sw6trgnw3h575f35gu/lib' '-DTPL_ENABLE_MPI:BOOL=ON' '-DCMAKE_C_COMPILER:STRING=/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.2.0/mpich-4.1.1-una3gsqpqddps5po56yrpbgyncipegdx/bin/mpicc' '-DCMAKE_CXX_COMPILER:STRING=/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.2.0/mpich-4.1.1-una3gsqpqddps5po56yrpbgyncipegdx/bin/mpic++' '-DCMAKE_Fortran_COMPILER:STRING=/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.2.0/mpich-4.1.1-una3gsqpqddps5po56yrpbgyncipegdx/bin/mpif90' '-DMPI_BASE_DIR:STRING=/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.2.0/mpich-4.1.1-una3gsqpqddps5po56yrpbgyncipegdx' '-DTPL_ENABLE_ParMETIS:BOOL=ON' '-DParMETIS_LIBRARY_DIRS:STRING=/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.2.0/parmetis-4.0.3-6kzi6romat5pn7jpmqz2v3v4irkr75s4/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.2.0/metis-5.1.0-cfyxpccu64h64r2nglgvxofktyitdspg/lib' '-DParMETIS_LIBRARY_NAMES:STRING=parmetis;metis' '-DTPL_ParMETIS_INCLUDE_DIRS:STRING=/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.2.0/parmetis-4.0.3-6kzi6romat5pn7jpmqz2v3v4irkr75s4/include;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.2.0/metis-5.1.0-cfyxpccu64h64r2nglgvxofktyitdspg/include' '-DHAVE_SUPERLUDIST_LUSTRUCTINIT_2ARG:BOOL=ON' '-DTPL_ENABLE_Cholmod:BOOL=OFF' '-DTeuchos_ENABLE_COMPLEX:BOOL=OFF' '-DTeuchos_ENABLE_FLOAT:BOOL=OFF' '-DTpetra_INST_OPENMP:BOOL=OFF' '-DTpetra_INST_DOUBLE:BOOL=ON' '-DTpetra_INST_COMPLEX_DOUBLE:BOOL=OFF' '-DTpetra_INST_COMPLEX_FLOAT:BOOL=OFF' '-DTpetra_INST_FLOAT:BOOL=OFF' '-DTpetra_INST_SERIAL:BOOL=ON' '-DTpetra_INST_INT_INT:BOOL=OFF' '-DTpetra_INST_INT_LONG:BOOL=OFF' '-DTpetra_INST_INT_LONG_LONG:BOOL=ON' '-DKokkos_ENABLE_CUDA:BOOL=OFF' '-DKokkos_ENABLE_OPENMP:BOOL=OFF' '-DKokkos_ENABLE_ROCM:BOOL=OFF' '-DKokkos_ENABLE_HIP:BOOL=ON' '-DKokkos_ENABLE_HIP_RELOCATABLE_DEVICE_CODE:BOOL=OFF' '-DTpetra_INST_HIP:BOOL=ON' '-DKokkos_ARCH_VEGA90A:BOOL=ON' '-DCMAKE_INSTALL_NAME_DIR:STRING=/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.2.0/trilinos-develop-6pqbqnx4d3tq3gjfqvjri5hn7ubivtdw/lib' '/tmp/root/spack-stage/spack-stage-trilinos-develop-6pqbqnx4d3tq3gjfqvjri5hn7ubivtdw/spack-src'

1 error found in build log:
     1176    --     Device Parallel: Kokkos::HIP
     1177    --     Host Parallel: NoTypeDefined
     1178    --       Host Serial: SERIAL
     1179    --
     1180    -- Architectures:
     1181    --  VEGA90A
  >> 1182    CMake Error at packages/kokkos/cmake/kokkos_tpls.cmake:29 (MESSAGE):
     1183      Enabled TPL HWLOC inside TriBITS build, but this can only be enabled in a
     1184      standalone build
     1185    Call Stack (most recent call first):
     1186      packages/kokkos/cmake/kokkos_tpls.cmake:34 (KOKKOS_TPL_OPTION)
     1187      packages/kokkos/cmake/kokkos_tribits.cmake:207 (INCLUDE)
     1188      packages/kokkos/CMakeLists.txt:220 (KOKKOS_SETUP_BUILD_ENVIRONMENT)
```


### Error message

<details><summary>Error message</summary>
<pre>
... see above
</pre></details>


### Information on your system

* **Spack:** 0.20.0.dev0 (3c3a4c75776ece43c95df46908dea026ac2a9276)
* **Python:** 3.8.10
* **Platform:** linux-ubuntu20.04-cascadelake
* **Concretizer:** clingo

### Additional information

[spack-build-01-cmake-out.txt](https://github.com/spack/spack/files/11453327/spack-build-01-cmake-out.txt)
[spack-build-env.txt](https://github.com/spack/spack/files/11453328/spack-build-env.txt)
[spack-build-out.txt](https://github.com/spack/spack/files/11453329/spack-build-out.txt)

@keitat @kuberry @psakievich @sethrj @jwillenbring @wspear

### General information

- [X] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [X] I have run `spack maintainers <name-of-the-package>` and **@mentioned** any maintainers
- [X] I have uploaded the build log and environment files
- [X] I have searched the issues of this repo and believe this is not a duplicate

--- a/var/spack/repos/builtin/packages/trilinos/package.py
+++ b/var/spack/repos/builtin/packages/trilinos/package.py
@@ -656,7 +656,7 @@ class Trilinos(CMakePackage, CudaPackage, ROCmPackage):
         )
 
         if spec.satisfies("@develop +stokhos"):
-            options.append("-DStokhos_ENABLE_PCE_Scalar_Type:BOOL=OFF")
+            options.append(self.define("Stokhos_ENABLE_PCE_Scalar_Type", False))
 
         if "+dtk" in spec:
             options.extend(

--- a/var/spack/repos/builtin/packages/trilinos/package.py
+++ b/var/spack/repos/builtin/packages/trilinos/package.py
@@ -655,6 +655,9 @@ class Trilinos(CMakePackage, CudaPackage, ROCmPackage):
             ]
         )
 
+        if spec.satisfies("@develop +stokhos"):
+            options.append("-DStokhos_ENABLE_PCE_Scalar_Type:BOOL=OFF")
+
         if "+dtk" in spec:
             options.extend(
                 [
@@ -770,7 +773,7 @@ class Trilinos(CMakePackage, CudaPackage, ROCmPackage):
         ]
         if spec.satisfies("@12.12.1:"):
             tpl_dep_map.append(("Pnetcdf", "parallel-netcdf"))
-        if spec.satisfies("@13:"):
+        if spec.satisfies("@13:") and not spec.satisfies("@develop"):
             tpl_dep_map.append(("HWLOC", "hwloc"))
 
         for tpl_name, dep_name in tpl_dep_map:

